### ISSUE TITLE
visual mode doesn't highlight selected text without a display server

### ISSUE THREAD

**Issue by pavlukivan** [neovim/neovim#15963]
### Neovim version (nvim -v)

0.5.1

### Vim (not Nvim) behaves the same?

no, vim 8.2.3441

### Operating system/version

Arch Linux

### Terminal name/version

Linux console

### $TERM environment variable

linux

### Installation

Official Arch repo

### How to reproduce the issue

Open any text without a display server, try highlighting it in visual mode

### Expected behavior

Text gets highlighted in the terminal (same as vim)

### Actual behavior

Text doesn't get highlighted, making it harder to use the visual mode

**Comment by zeertzjq** [comment]
Neovim's default colorscheme is very old. Maybe `cterm=reverse` is needed for `Visual`.

**Comment by clason** [comment]
Just to make sure: you are talking about the standard Linux console you get when you are booting into runlevel 3 or below? If so, please add more details on your system (Linux kernel version etc.)

**Comment by justinmk** [comment]
> Nvim uses 256 colours by default, ignoring terminfo for most terminal types, including "linux" (whose virtual terminals have had 256-colour support since 4.8)

yep. but see also the commit messages in https://github.com/neovim/neovim/pull/5001 . 

Open to improving edge cases, but note that we absolutely favor compromises for terminal quirks, if it makes startup more deterministic (which matters for remote UIs): https://github.com/neovim/neovim/pull/9829

**Comment by pavlukivan** [comment]
> or a faulty assumption on Neovim's side

That seems to be it. Linux kernel does "support" 256 colors, however it simply downgrades the color depth.

**Comment by justinmk** [comment]
Maybe we just need to adjust the Visual highlight so that it downgrades to something more reasonable? We want to change the default colorscheme anyway. Default `Visual` should just be "white" for bg=dark, and "black" for bg=light.

**Comment by zeertzjq** [comment]
The new default colorscheme doesn't have this issue, but `&t_Co` still seems incorrect in Linux Console (it should be 8, but it is 256).

**Comment by scy** [comment]
Yep. The problem here is an unclear definition of what "supports 256 colors" means.

Does the Linux console _interpret_ 256-color values? Yes.

Does it _render_ 256 colors? No, it maps them down to its 16-color palette (backgrounds are even 8-colors only).

So, a script or colorscheme that tries to "progressively enhance" based on the terminal's capabilities, it doesn't matter whether the terminal can understand the escape sequences; it matters whether it actually displays things in the intended way.

The Linux console does not display 256 colors, so it should be set to "supports 16 foreground and 8 background colors", if that's possible.

Since `&t_Co` is used by quite a few scripts & colorschemes to dynamically change their behavior, but writing to `&t_Co` is not allowed, we can't even add a workaround like

```vim
if $TERM == 'linux'
    let &t_Co = 16
endif
```

to our config. :(

**Comment by GaKu999** [comment]
> Does it _render_ 256 colors? No, it maps them down to its 16-color palette (backgrounds are even 8-colors only).

If one reaaally wants the linux console to do bright backgrounds, it uses blink for that, for **some** reason.
